pyqtorch/core/utils.py

```diff
@@ -150,7 +152,7 @@ def _apply_batch_gate(
     Same shape as `input_state`
     """
     mat = mat.reshape([2] * len(qubits) * 2 + [batch_size])
-    qubits = N_qubits - 1 - np.array(qubits)
+    qubits = np.array(N_qubits - 1 - np.array(qubits), dtype=np.int64)
```

Comment:
why numpy array here?

Reply:
Well, to specify the `dtype`. Without it, it defaults the `dtype` to objects as it can't infer it from `qubits`.